lms/djangoapps/grades/models.py

```diff
@@ -243,6 +244,7 @@ def save_grade(cls, **kwargs):
         except IntegrityError:
```

Comment:
Can you update the `get_or_create` query call?  Since we're not passing `course_id` it won't use the index.